src/System.Runtime.Extensions/tests/Performance/System/Perf.Convert.cs

```diff
@@ -0,0 +1,1982 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Xunit.Performance;
+using Xunit;
+
+namespace System
+{
+    public class Perf_Convert
+    {
+        [Benchmark(InnerIterationCount = 100000)]
+        [InlineData(1024, 1408, Base64FormattingOptions.InsertLineBreaks)]
+        [InlineData(1024, 1368, Base64FormattingOptions.None)]
+        public void ToBase64CharArray(int binaryDataSize, int arraySize, Base64FormattingOptions formattingOptions)
+        {
+            int innerIterationCount = (int)Benchmark.InnerIterationCount;
+            byte[] binaryData = new byte[binaryDataSize];
+            char[] base64CharArray = new char[arraySize];
+
+            foreach (var iteration in Benchmark.Iterations)
+            {
+                using (iteration.StartMeasurement())
+                {
+                    for (int i = 0; i < innerIterationCount; i++)
+                    {
+                        Convert.ToBase64CharArray(binaryData, 0, binaryDataSize, base64CharArray, 0, formattingOptions);
+                    }
+                }
+            }
+        }
+
+        [Benchmark(InnerIterationCount = 100000)]
+        [InlineData(Base64FormattingOptions.InsertLineBreaks)]
+        [InlineData(Base64FormattingOptions.None)]
+        public void ToBase64String(Base64FormattingOptions formattingOptions)
+        {
+            int innerIterationCount = (int)Benchmark.InnerIterationCount;
+            byte[] binaryData = new byte[1024];
+
+            foreach (var iteration in Benchmark.Iterations)
+            {
+                using (iteration.StartMeasurement())
+                {
+                    for (int i = 0; i < innerIterationCount; i++)
+                    {
+                        Convert.ToBase64String(binaryData, formattingOptions);
+                    }
+                }
+            }
+        }
+
+        [Benchmark(InnerIterationCount = 380000000)]
+        public void ToBoolean_Boolean()
+        {
+            int innerIterationCount = (int)Benchmark.InnerIterationCount;
+
+            foreach (var iteration in Benchmark.Iterations)
+            {
+                using (iteration.StartMeasurement())
+                {
+                    for (int i = 0; i < innerIterationCount; i++)
+                    {
+                        Convert.ToBoolean(true);
```

Comment:
Have you looked at the asm for this?  I would expect the JIT would inline this call, at which point it would evaporate, since the implementation of this method is just `return value;`.

Reply:
(Similar concerns for a bunch of the rest of these To tests)

Reply:
I will remove all the test that are returning just the values. I wasn’t in 100% sure if they are pointless....

Reply:
@KostaVlev, it's not just those.  Many of these Convert methods are purely casts, or branches that can be resolved at compile time when the JIT inlines the method.  For example, you've added test for Convert.ToInt32(bool) here:
https://github.com/dotnet/corefx/pull/24793/files#diff-e74efccef59eb3ac2f6b3e9a18750273R750
but all of the work in that loop is going to go away in a release build when that ToInt32 gets inlined.  As an example, this C#:
```
using System;

class Program
{
    static void Main()
    {
        int d = 0;
        for (int i = 0; i < 100; i++) d = Convert.ToInt32(true);
        Console.WriteLine(d);
    }
}
```
results in this asm for Main:
```
G_M45812_IG01:
       4883EC28             sub      rsp, 40

G_M45812_IG02:
       33C9                 xor      ecx, ecx

G_M45812_IG03:
       B801000000           mov      eax, 1
       FFC1                 inc      ecx
       83F964               cmp      ecx, 100
       7CF4                 jl       SHORT G_M45812_IG03

G_M45812_IG04:
       8BC8                 mov      ecx, eax
       E80FF5FFFF           call     System.Console:WriteLine(int)
       90                   nop

G_M45812_IG05:
       4883C428             add      rsp, 40
       C3                   ret
```
You can see that loop at G_M45812_IG03 is simply resulting in `1` being stored into `eax` (the register for `d`) 100 times

Reply:
That seems valid also for methods that returns value != 0 for example since it is going to be resolved to 1 or 0? I will submit clean up shortly ...